quarkchain/cluster/tests/test_cluster.py

```diff
@@ -48,15 +48,32 @@ def _tip_gen(shard_state):
 
 
 class MockGrpcServer(grpc_pb2_grpc.ClusterSlaveServicer):
-    def __init__(self):
+    def __init__(self, expected_minor_block_headers):
         self.request_num = 0
+        self.expected_minor_block_headers = (
+            expected_minor_block_headers
+        )  # type: List[MinorBlockHeader]
 
     def SetRootChainConfirmedBlock(self, request, context):
         self.request_num += 1
         return grpc_pb2.SetRootChainConfirmedBlockResponse(
             status=grpc_pb2.ClusterSlaveStatus(code=0, message="Test")
         )
 
+    def AddRootBlock(self, request, context):
+        assert len(self.expected_minor_block_headers) == len(
```

Comment:
this test should be different with the one testing grpc client: we don't care about what `grpc_client.add_root_block` does, we only care about whether it's been called or not. so I suggest just using `request_num` to record number of invocations

Reply:
Done.